cilium/CFP-32759-cilium-mtls-via-mtls-agent.md

```diff
@@ -0,0 +1,260 @@
+
+When the packet reaches mTLS Agent B’s pod network namespace, TPROXY rules in the mTLS agent network namespace will match the packet based on its destination port 15008 and redirect it to 127.0.0.1:15008 via the local loopback device. The packet will reach the process listening on the mTLS Agent at port 15008.
+
+![alt text](images/cilium-mTLS-datapath-step-2.png)
+
+#### Step 3: From mTLS Agent B to Pod B
+
+When the packet reaches mTLS Agent B, we are able to read the request’s header and get the original destination port. If not already established, a new TCP session is established with Pod B at the original destination port. The packet is then routed from mTLS Agent B to Pod B using the standard routing rules of the cluster.
+
+![alt text](images/cilium-mTLS-datapath-step-3.png)
+
+## Requirements
+
+The targeted Cilium release for an MVP of mTLS via transparent proxying is v1.17. 
+
+#### MVP Components
+1. Transparent Redirection
+1. Cilium xDS Server 
+1. mTLS via Transparent Proxy
+
+### MVP Scope
+
+Users can enable Cilium mTLS per workload. All traffic leaving a node from an mTLS-enabled source destined for an mTLS-enabled destination on a different node will be sent over mTLS.
+
+This proposal is scoped to the following traffic configurations:
+* Traffic between workloads on the same node will not be proxied (and therefore not be encrypted)
+* Cross-node traffic on a single cluster is tunneled over mTLS
+* Node egress traffic to non-mTLS-enabled destinations is not proxied
+
+Out of scope configurations (design should be extensible to support these configuration, but they are out of scope for this proposal):
+* Multi-cluster traffic
+
+### Enablement API
+
+For an MVP, the mTLS agent functionality will be enabled in Cilium via an environment variable in the cilium configuration - _mtlsAgent: enable_. The Cilium agent will identify the mTLS agent via a pod label - _cilium.io/redir-type=proxy_. Workloads are enabled for redirection and identified by the Cilium agent with the following label - _cilium.io/redir-type=workload_. For the MVP, the user will apply the labels per workload. A future goal is to support enabling workloads by namespace with support for exclusions. The Cilium agent will use these labels to identify which pods should be added to the BPF redirection map and to validate requests for workload configuration.
+
+### Installation
+
+The mTLS agent will be installed and managed separately from Cilium. The order of installation for the mTLS agent and Cilium is not significant. Cilium must be installed or configured with the mTLS agent functionality enabled. The mTLS agent will be deployed separately. Once the mTLS agent and cilium are configured for cilium mTLS, all existing workloads with existing _cilium.io/redir-type=workload_ labels will be enabled for redirection.
```

Comment:
If need be, it is possible to have the mTLS agent included as part of the Cilium install, similarly to how we include SPIRE in the current mutual auth solution.